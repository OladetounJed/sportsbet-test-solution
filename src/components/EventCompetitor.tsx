import React from "react";

interface Props {
  competitor: {
    score: number;
    name: string;
  };
}

export function EventCompetitor(props: Props) {
  return (
    <div className=" flex items-center justify-between text-trunks">
      <p
        className="font-light text-xs sm:text-sm text-white"
      >
        {props.competitor?.name}
      </p>
      <p className="text-piccolo text-xs sm:text-sm" data-testid="competitor-score">
        {props.competitor?.score}
      </p>
    </div>
  );
}
