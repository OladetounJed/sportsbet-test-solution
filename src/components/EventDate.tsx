import React from "react";

interface Props  {
  date: string;
  time: string;
};

export function EventDate(props: Props) {
  return (

      <div className=" flex text-xs sm:text-sm text-trunks gap-1">
        <p>{props.date } {props.time}</p>
      </div>

  );
}
