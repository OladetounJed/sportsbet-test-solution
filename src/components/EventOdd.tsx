import React from "react";

interface Props {
  selection: { odds?: Number };
}

export function EventOdd(props: Props) {
  return (
    <div
      data-testid="event-odd"
      onClick={() => {
        props.selection?.odds ? console.log(props.selection) : null;
      }}
      className={
        "h-full bg-goku flex items-center justify-center text-white text-xs sm:text-base  font-semibold w-full rounded" +
        (props.selection?.odds ? " cursor-pointer" : " cursor-not-allowed")
      }
    >
      {props.selection?.odds?.toFixed(2) || "-"}
    </div>
  );
}
