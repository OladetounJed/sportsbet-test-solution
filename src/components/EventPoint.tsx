import React from "react";

export function EventPoint() {
  return (
    <div
      className={
        "h-full bg-goku flex items-center justify-center text-piccolo text-xs sm:text-base   font-semibold w-full rounded"
      }
    >
      +92
    </div>
  );
}
