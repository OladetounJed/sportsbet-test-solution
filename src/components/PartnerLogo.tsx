import Link from "next/link";
import React from "react";

interface Props {
  href?: string;
  icon: JSX.Element;
}

export function PartnerLogo(props: Props) {
  return (
    <div role="partner-logo" className="px-3 py-2 bg-plank rounded-full flex items-center justify-center">
      {props.icon}
    </div>
  );
}
