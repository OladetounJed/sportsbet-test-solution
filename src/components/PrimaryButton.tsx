import React from "react";

interface Props {
  type?: "button" | "submit" | "reset";
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  disabled?: boolean;
  text: string;
}

export function PrimaryButton(props: Props) {
  return (
    <button
      className="bg-piccolo text-white text-base sm:text-xl  font-normal rounded-full tracking-wider p-3 sm:p-5 my-4 w-full"
      onClick={props.onClick}
      type={props.type}
      disabled={props.disabled}
    >
      {props.text}
    </button>
  );
}
