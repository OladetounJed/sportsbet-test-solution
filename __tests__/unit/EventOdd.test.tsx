import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { EventOdd } from "@/components/EventOdd";

test("renders event odds component correctly", async () => {
  const {container} = render(
    <EventOdd selection={{ odds: 12 }} />
  );
  
  expect(container.firstChild).toMatchSnapshot()
  expect(screen.getByTestId("event-odd")).toHaveTextContent("12");
});

test("check if odd component doesn't click when there is no odd", async () => {
  render(<EventOdd selection={{ odds: undefined }} />);
  const handleClick = jest.fn();
  fireEvent.click(screen.getByTestId("event-odd"));
  expect(handleClick).toHaveBeenCalledTimes(0);
});
