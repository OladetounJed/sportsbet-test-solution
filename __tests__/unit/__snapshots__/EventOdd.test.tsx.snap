// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders event odds component correctly 1`] = `
<div
  class="h-full bg-goku flex items-center justify-center text-white   font-semibold w-full rounded cursor-pointer"
  data-testid="event-odd"
  style="opacity: 0; transform: none;"
>
  12.00
</div>
`;
